cmake/modules/FindMaterialX.cmake

```diff
@@ -0,0 +1,229 @@
+#
+# TM & (c) 2017 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+# All rights reserved.  See LICENSE.txt for license.
+#
+# Helper CMake module to Find MaterialX include dirs, libraries and document libraries
+#
+# Example Usage
+#  - Build and Install MaterialX 
+#  - register the path to this cmake module using CMAKE_MODULE_PATH
+#  - use find_package(MaterialX REQUIRED) to locate core components
+#  - additonal components can be found using shaderx, bxdf and pytton
+#     e.g:  find_package(MaterialX REQUIRED COMPONENTS shaderx)
+#           will find inc, lib, corelib and generators
+#
+# Variables defined:
+# MATERIALX_FOUND            True if headers and requested libraries were found
+# MATERIALX_BASE_DIR         MaterialX root installation directory
+# MATERIALX_INCLUDE_DIRS     MaterialX include directory
+# MATERIALX_LIB_DIRS         MaterialX lib directory
+# MATERIALX_CORE_LIBS        MaterialX Core libraries and ShaderX i.e. Core, Format & GenShader
+# MATERIALX_GENERATOR_LIBS   MaterialX Generator libraries i.e GenGlsl, GenOsl
+# MATERIALX_RENDER_LIBS      MaterialX Render libraries i.e RenderGlsl, RenderOsl
+# MATERIALX_STDLIB_DIR       Path to the MaterialX standard library directory
+# MATERIALX_PBRLIB_DIR       Path to the MaterialX pbr library directory
+# MATERIALX_BXDFLIB_DIR      Path to the Surface shaders library directory e.g. standard surface
+# MATERIALX_PYTHON_DIR       Path to MaterialX Python library
+# MATERIALX_RESOURCES_DIR    Path to MaterialX Resources (sample data, mtlx etc)
+#
+
+# make core variables required
+set (MATERIALX_REQUIRED_VARS
+        MATERIALX_INCLUDE_DIRS
+        MATERIALX_LIB_DIRS
+        MATERIALX_STDLIB_DIR
+        MATERIALX_CORE_LIBS
+    )
+
+# make shaderx a required if requested
+if ("shaderx" IN_LIST MaterialX_FIND_COMPONENTS)
```

Comment:
Now that the ShaderX functionality is merged into MaterialX, I would recommend that we refer to these features by their unique functionality (e.g. shader code generation, PBR nodes).  Would "shadergen" be a reasonable string to check for here?

Reply:
"shadergen" sounds good as that is the prefix used for all generation modules.

Reply:
I think this is the one remaining change we need before merging, unless we'd like to include additional improvements in this PR.

Reply:
Forgot to fix this. Update added now.